### ISSUE TITLE
Unable to interact with modded blocks

### ISSUE THREAD

**Issue by A-wolf** [chimericdream/hopper-xtreme#5]
**Describe the bug**
Does not interact with any modded blocks 

**To Reproduce**
1. Open game
2. Put the hopper into or out of a block such as a sophisticated storage barrel 
3. Put item into the hopper
4. Nothing moves

**Expected behavior**
Items moving from the hopper into the storage as a vanilla hopper does

**Screenshots**

**Environment**
 * Minecraft version: 1.21.1
 * Mod version: [NeoForge] 1.21.x-2.1.0 [https://www.curseforge.com/minecraft/mc-mods/hopper-xtreme/files/5757579]
 * OS: windows 11 pro

**Additional context**
Tried some other blocks too and pulling out of but nothing modded interacted with it

**Comment by chimericdream** [comment]
Sorry for my delayed response on this. I will look into this and see what I can figure out. Thanks for the report!

**Comment by A-wolf** [comment]
No worries don't stress about taking your time :) I had already moved onto another solution installed in my modpack but figured I still mention it as you had a cool mod :)

**Comment by chimericdream** [comment]
I just tried with both Minekea (another of my mods) and Sophisticated Storage. Minekea's modded containers worked fine, but the barrels from SS didn't. It seems like the other dev might be doing their own kind of inventory management, but I am not sure. I definitely noticed that even some vanilla behavior (middle-click, middle-click with drag, etc) doesn't work in the SS barrel, so my gut tells me the issue is on their end.

I'll leave this open for now, in case there are other mods besides Sophisticated Storage that you were having issues with, but in that case at least, it seems to be a compat issue on their side.

**Comment by PurpleHel** [comment]
Same problem. 1.21.1, hopper xtreme 2.2.1, neoforge
hopper xtreme absolutely doesn't work with sophisticated storage or sophisticated backpacks. Vanilla hoppers work fine, pipes from 3 different mods all work, hopper xtreme does not work. I'm not sure it's a sophisticated $thing issue?

**Comment by chimericdream** [comment]
> Same problem. 1.21.1, hopper xtreme 2.2.1, neoforge hopper xtreme absolutely doesn't work with sophisticated storage or sophisticated backpacks. Vanilla hoppers work fine, pipes from 3 different mods all work, hopper xtreme does not work. I'm not sure it's a sophisticated $thing issue?

Which mods' pipes work with the sophisticated stuff? It's possible they have some special handling for sophisticated storage, so if I can see what someone else is doing I may be able to replicate it.

For my part, I am just checking if the block implements `InventoryProvider` or if the block entity implements `Inventory`. I'd be happy to add support if I can, but sophisticated storage doesn't use either of those interfaces as far as I have been able to determine.

**Comment by P3pp3rF1y** [comment]
Someone left a note on my cf page that hopper doesn't work with SS. Just briefly checked the code and I don't believe that I see any `IItemHandler` capability implementation anywhere, which was SS and I believe anything else in neoforge ecosystem uses also Neoforge wraps all vanilla inventories with this capability so that can be universally used to access any inventory and hence is the way to go when working with items. 
https://docs.neoforged.net/docs/datastorage/capabilities

**Comment by chimericdream** [comment]
> Someone left a note on my cf page that hopper doesn't work with SS. Just briefly checked the code and I don't believe that I see any `IItemHandler` capability implementation anywhere, which was SS and I believe anything else in neoforge ecosystem uses also Neoforge wraps all vanilla inventories with this capability so that can be universally used to access any inventory and hence is the way to go when working with items. https://docs.neoforged.net/docs/datastorage/capabilities

Yeah, that'd certainly explain things. I appreciate you taking the time to let me know! I should be able to set aside time this week to work on a fix, so feel free to let your user know that I am working on it.
